src/contexts/RecordingSelectionContext.ts

```diff
@@ -224,6 +243,36 @@ const panTime = (state: RecordingSelection, action: PanRecordingSelectionAction)
     return {...state, visibleTimeStartSeconds: newStart, visibleTimeEndSeconds: newEnd, focusTimeSeconds: focus }
 }
 
+const panTimeDeltaT = (state: RecordingSelection, action: PanRecordingSelectionDeltaTAction): RecordingSelection => {
+    if (state.visibleTimeStartSeconds === false || state.visibleTimeEndSeconds === false || state.recordingStartTimeSeconds === false || state.recordingEndTimeSeconds === false) {
+        console.warn(`WARNING: Attempt to call panTime() with uninitialized state ${state}.`)
+        return state
+    }
+    const windowLength = state.visibleTimeEndSeconds - state.visibleTimeStartSeconds
+    const panDisplacementSeconds = action.deltaT
+    let newStart = state.visibleTimeStartSeconds    
+    let newEnd = state.visibleTimeEndSeconds
+    if (panDisplacementSeconds > 0) {
+        // panning forward. Just need to check that we don't run over the end of the recording.
+        newEnd = Math.min(state.visibleTimeEndSeconds + panDisplacementSeconds, state.recordingEndTimeSeconds)
+        newStart = Math.max(newEnd - windowLength, state.recordingStartTimeSeconds)
+    } else if (panDisplacementSeconds < 0) {
+        // panning backward. Need to make sure not to put the window start time before the recording start time.
+        newStart = Math.max(state.visibleTimeStartSeconds + panDisplacementSeconds, state.recordingStartTimeSeconds)
+        newEnd = Math.min(newStart + windowLength, state.recordingEndTimeSeconds)
+    } else {
+        return state
+    }
+    const keepFocus = state.focusTimeSeconds && state.focusTimeSeconds > newStart && state.focusTimeSeconds < newEnd
+    const focus = keepFocus ? state.focusTimeSeconds : undefined
+
+    // Avoid creating new object if we didn't actually change anything
+    if (newStart === state.visibleTimeStartSeconds && newEnd === state.visibleTimeEndSeconds) return state
+
+    // console.log(`Returning new state: ${newStart} - ${newEnd} (was ${state.visibleTimeStartSeconds} - ${state.visibleTimeEndSeconds})`)
+    return {...state, visibleTimeStartSeconds: newStart, visibleTimeEndSeconds: newEnd, focusTimeSeconds: focus }
```

Comment:
Since this logic is repeated for both state changes (pan by pct and pan by defined amount) it might be a good idea to pop it out into a function that both verbs call.

Reply:
I created a helper function for this as you suggested